### PR TITLE
Dataview sync fixes

### PR DESCRIPTION
**Pull request by milesAraya**
## Summary

- Add one-time startup sync so API worker containers download published experiments on boot (not just from the periodic scheduler)
- Handle empty EC2 Reservations in orphaned data cleanup to prevent KeyError crashes
- Migrate raw SQL to SQLAlchemy ORM in the storage reconciliation job
- Force-rebuild Terraform Lambda package on every apply (temporary debug trigger)

---

### `studio/__main_unit__.py`
- Schedule a background `asyncio.create_task` at startup that waits 5s then calls `PublishedExperimentSyncJob.run_startup_sync()`
- Runs on ALL non-standalone containers, even those with the scheduler disabled, ensuring published experiments are available locally

### `studio/app/common/core/background/sync_job.py`
- **`run_startup_sync()`**: New classmethod that queries all published experiments, checks local file presence, and downloads missing ones in two phases (thumbnails at high concurrency, then metadata at lower concurrency)
- **`_get_all_published_experiments()`**: New classmethod that queries all published+successful experiments regardless of `local_sync_status`, used by the startup sync

### `studio/app/common/core/background/cleanup_job.py`
- Use `.get()` for EC2 `describe_instances` response instead of direct dict indexing
- Handle empty `Reservations` list (or missing `Instances` key) by treating the instance as terminated and cleaning up the orphaned assignment

### `studio/app/common/core/background/storage_reconciliation_job.py`
- Replace all raw SQL strings with SQLAlchemy ORM queries using `select()`, `func.count()`, and `or_()`
- Import `UserStorageUsage` model and use typed column access instead of string-based SQL

### `infrastructure/terraform/common_user_manager.tf`
- Change Lambda package rebuild trigger from `filesha256()` (content-based) to `timestamp()` (always-run) -- forces dependency reinstall on every `terraform apply`

### Tests
- **`test_cleanup_job.py`**: Add `test_handle_orphaned_data_empty_reservations` covering the empty EC2 Reservations edge case
- **`test_sync_job.py`**: Add `TestStartupSync` class with 3 tests: downloads missing experiments, skips locally present experiments, handles empty published list

---

## Behavior Changes

```
| Scenario                        | Before                             | After                                    |
|---------------------------------|------------------------------------|------------------------------------------|
| API worker container startup    | No sync until scheduler runs       | Startup sync downloads missing published |
|                                 |                                    | experiments after 5s delay               |
| EC2 returns empty Reservations  | KeyError crash in cleanup job      | Treated as terminated, cleanup proceeds  |
| Storage reconciliation queries  | Raw SQL strings                    | SQLAlchemy ORM select() statements       |
| Terraform Lambda rebuild        | Only when code hash changes        | Every terraform apply (temporary)        |
```

---

## Risk Assessment

```
| Area                  | Risk   | Mitigation                                              |
|-----------------------|--------|---------------------------------------------------------|
| Startup sync          | Medium | 5s delay avoids boot contention; exceptions are caught  |
|                       |        | and logged; does not modify DB sync status              |
| EC2 empty Reservations| Low    | Defensive .get() with fallback; matches existing        |
|                       |        | InvalidInstanceID.NotFound handling pattern             |
| ORM migration         | Low    | Functionally equivalent queries; no schema changes      |
| Terraform trigger     | Low    | Only affects build speed; should be reverted before     |
|                       |        | production                                              |
```

### CODE REVIEW

File: studio/__main_unit__.py [hunk 1]
```diff
@@ -95,6 +95,23 @@ async def lifespan(app: FastAPI):
     # (e.g., when using cron)
     disable_scheduler = os.environ.get("DISABLE_BACKGROUND_SCHEDULER", "0") == "1"
 
+    # Startup sync runs on ALL containers (including API workers
```
Comment (itutu-tienday) [inline studio/__main_unit__.py:98]: Could you please explain the details of the bug and the scenario in which it occurred that prompted you to apply this fix?
*Please include this in the PR description.

Comment (milesAraya) [reply studio/__main_unit__.py:98]: Explanation added to https://github.com/arayabrain/optinist-for-cloud/pull/311
